### PR TITLE
chore(repo): Add turborepo

### PR DESCRIPTION
**Pull request by igneel64**
Minor Lerna additions

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [x] other:

## Packages affected

- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [x] `tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

- Naive [Turborepo](https://turborepo.org/docs/guides/migrate-from-lerna) addition. It caches results from tests in packages that have not changed. Same with the build process(_understands inter-package dependencies_).
- Lerna allowed branches config.

(_Let's validate on the next deployment that the distribution files are receiving the correct `package.json version` after the `version` command_)

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -5,7 +5,18 @@
     ],
     "command": {
         "version": {
-            "message": "chore(release): Publish"
+            "message": "chore(release): Publish",
+            "allowBranch": [
```
Comment (igneel64) [inline lerna.json:9]: This will bring a bit of safety around releasing only in `main` or `next` branches.

**Comment by nikosdouvlis** [comment]
Damn it @igneel64 I wanted to play with turborepo and add it next week, was hoping you forgot about it 😩

Did a quick review, looks good! Will test more tomorrow. Also, cant wait for the `@next` releases. Nice job

**Review by gkats — APPROVED** [review]

**Comment by SokratisVidros** [comment]
I am a bit confused. Since we have lerna why do we need this?

**Comment by igneel64** [comment]
@SokratisVidros 
- The 'generic' reasoning can be found here: https://turborepo.org/docs/guides/migrate-from-lerna

- The reason for integration as mentioned in TODOs `Add turborepo for performance optimizations.`. Lerna does not have the capability to cache task results. That means that when you try to test or build or your packages, it all happens from scratch. With turborepo, the most basic addition is that if the file hashes for a package are the same, the result is cached. E.g. if you make changes only to the tests of clerk-sdk-node and run again all the package tests, clerk-sdk-node only will need to be re-tested, the other results are returned from cache. Same thing with build. This saves quite a lot amount of time.

- Moving forward with more packages we will also benefit from the task scheduling and parallelization capabilities or turborepo.
image

**Pull request merged**